File: xbonacci_sequence/xbonacci_sequence.py
from itertools import chain, islice


def xbon(signature):
    while True:
        k = sum(signature)
        for i in range(len(signature) - 1):
            signature[i] = signature[i + 1]
        signature[-1] = k
        yield k


def Xbonacci(signature, n):
    xbon_gen = xbon(signature)
    return list(islice(chain(signature, xbon_gen), n))

File: xbonacci_sequence/xbonacci_sequence.go
package kata

func Xbon(signature []int) chan int {
	ch := make(chan int)
	go func() {
		for true {
			k := 0
			for _, cur := range signature {
				k += cur
			}
			for i := 0; i < len(signature)-1; i++ {
				signature[i] = signature[i+1]
			}
			signature[len(signature)-1] = k
			ch <- k
		}
	}()
	return ch
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func Xbonacci(signature []int, n int) []int {
	res := make([]int, min(len(signature), n))
	copy(res, signature)
	xbon_gen := Xbon(signature)
	for i := len(signature); i < n; i++ {
		res = append(res, <-xbon_gen)
	}
	return res
}
